MaterialDesignThemes.Wpf/ScrollViewerAssist.cs

```diff
@@ -1,169 +1,254 @@
-            "ShowSeparators", typeof(bool), typeof(ScrollViewerAssist), new PropertyMetadata(default(bool)));
+    public static void SetShowSeparators(DependencyObject element, bool value)
+    {
+        element.SetValue(ShowSeparatorsProperty, value);
+    }
 
-        public static void SetShowSeparators(DependencyObject element, bool value)
-        {
-            element.SetValue(ShowSeparatorsProperty, value);
-        }
+    public static bool GetShowSeparators(DependencyObject element)
+    {
+        return (bool)element.GetValue(ShowSeparatorsProperty);
+    }
 
-        public static bool GetShowSeparators(DependencyObject element)
-        {
-            return (bool)element.GetValue(ShowSeparatorsProperty);
-        }
+    public static readonly DependencyProperty IgnorePaddingProperty = DependencyProperty.RegisterAttached(
+        "IgnorePadding", typeof(bool), typeof(ScrollViewerAssist), new PropertyMetadata(true));
 
-        public static readonly DependencyProperty IgnorePaddingProperty = DependencyProperty.RegisterAttached(
-            "IgnorePadding", typeof(bool), typeof(ScrollViewerAssist), new PropertyMetadata(true));
+    public static void SetIgnorePadding(DependencyObject element, bool value) => element.SetValue(IgnorePaddingProperty, value);
+    public static bool GetIgnorePadding(DependencyObject element) => (bool)element.GetValue(IgnorePaddingProperty);
 
-        public static void SetIgnorePadding(DependencyObject element, bool value) => element.SetValue(IgnorePaddingProperty, value);
-        public static bool GetIgnorePadding(DependencyObject element) => (bool)element.GetValue(IgnorePaddingProperty);
+    private static readonly DependencyProperty HorizontalScrollHookProperty = DependencyProperty.RegisterAttached(
+        "HorizontalScrollHook", typeof(HwndSourceHook), typeof(ScrollViewerAssist), new PropertyMetadata(null));
 
-        private static readonly DependencyProperty HorizontalScrollHookProperty = DependencyProperty.RegisterAttached(
-            "HorizontalScrollHook", typeof(HwndSourceHook), typeof(ScrollViewerAssist), new PropertyMetadata(null));
+    public static readonly DependencyProperty SupportHorizontalScrollProperty = DependencyProperty.RegisterAttached(
+        "SupportHorizontalScroll", typeof(bool), typeof(ScrollViewerAssist), new PropertyMetadata(false, OnSupportHorizontalScrollChanged));
 
-        public static readonly DependencyProperty SupportHorizontalScrollProperty = DependencyProperty.RegisterAttached(
-            "SupportHorizontalScroll", typeof(bool), typeof(ScrollViewerAssist), new PropertyMetadata(false, OnSupportHorizontalScrollChanged));
+    public static readonly DependencyProperty IgnoreVerticalScrollProperty = DependencyProperty.RegisterAttached(
+        "IgnoreVerticalScroll", typeof(bool), typeof(ScrollViewerAssist), new PropertyMetadata(false, OnIgnoreVerticalScrollChanged));
```

Comment:
I would move this down so that it is next to `OnIgnoreVerticalScrollChanged`

Reply:
Fair point; will move it.

Also, I am actually not satisfied with the name of it. Because in reality it does not ignore the vertical scroll, it simply re-raises the event allowing it to bubble further up the visual tree. Suggestions for a good name are welcome :-)

I have now renamed it to `RaiseVerticalScrollUpstream` but was also considering `BubbleVerticalScrollUpwards`. Thoughts?

Reply:
What about `BubbleVerticalScroll`? Given that routed events using bubbling to indicate that an event flow upward through the tree, I think that the direction is clear.

Reply:
Yeah I guess that makes sense.
My only concern is that it does not actually bubble the **original** event (because that is already handled), instead it creates a copy; this was the reasoning for going with the "Raise" prefix rather than the bubble.

Reply:
That is fair, though I think the "Raise" prefix might have people assuming they can register for the event on the ScrollViewer itself which would then miss the event since it starts at the ScrollViewer's parent rather than on the ScrollViewer itself.

I think I will merge this for now, but happy to revisit alternate names.